kod/object/active/holder/room/ghall/waylayoasis.kod

```diff
@@ -0,0 +1,418 @@
+      
+      plInside_doors = [ [68,106], [69,106], [70,106], [71,106], [72,106]];
+
+      % This is a box containing the guild door, used
+      % for removing players from it when it closes.
+      plEntrance = [ 6850, 7250, 10650, 11100 ];
+
+      propagate;
+   }
+ 
+   CreateStandardExits()
+   {
+      plEdge_Exits = $;
+      plEdge_Exits = Cons([ LEAVE_WEST, RID_DESERTPATH1, 11, 58, ROTATE_NONE ],plEdge_Exits);
+      plEdge_Exits = Cons([ LEAVE_EAST, RID_DESERTBRIDGE, 4, 5, ANGLE_EAST ],plEdge_Exits);
+
+      propagate;
+   }
+
+   SomethingTryGo(what=$, row=$,  col = $)
+   {
+      local i;
+      
+      foreach i in plInside_doors
+      {
+         if row = First(i)
+            AND col = Nth(i,2)
+         {
+            Send(self,@OpenEntrancedoor);
+
+            return TRUE;
+         }
+      }
+
+      propagate;
+   }
+
+
+   InFoyer(who=$,iRow=$,iCol=$,iFineRow=$,iFineCol=$)
+   {
+      if Send(self,@GetSection,#who=who,#iRow=iRow,#iCol=iCol,
+            #iFineRow=iFineRow,#iFineCol=iFineCol) = AREA_OUTSIDE
+      {
+         return TRUE;
+      }
+
+      return FALSE;
+   }
+
+   GetSection(who=$,iRow=$,iCol=$,iFineRow=$,iFineCol=$)
+   {
+      if who <> $
+      {
+         if Send(who,@GetOwner) <> self
+         {
+            return FALSE;
+         }
+
+         iRow = Send(who,@GetRow);
+         iCol = Send(who,@GetCol);
+         iFineCol = Send(who,@GetFineCol);
+         iFineRow = Send(who,@GetFineRow);
+      }
+
+      if iRow >= InsideAreaSouthMinRow
+         AND iFineRow >= InsideAreaSouthMinFineRow
+         AND iRow <= InsideAreaSouthMaxRow
+         AND iFineRow <= InsideAreaSouthMaxFineRow
+         AND iCol >= InsideAreaSouthMinCol
+         AND iFineCol >= InsideAreaSouthMinFineCol
+         AND iCol <= InsideAreaSouthMaxCol
+         AND iFineCol <= InsideAreaSouthMaxFineCol
+      {
+         return AREA_INSIDE_SOUTH;
+      }
+
+      if iRow >= InsideAreaSouthHallwayMinRow
+         AND iRow <= InsideAreaSouthHallwayMaxRow
+         AND iCol >= InsideAreaSouthHallwayMinCol
+         AND iCol <= InsideAreaSouthHallwayMaxCol
+      {
+         return AREA_INSIDE_SOUTH_HALLWAY;
+      }
+      
+      if Send(who,@GetHeightAtObject) = 560
+         OR Send(who,@GetHeightAtObject) = 600
+      {
+         return AREA_SOUTH_LEDGES;
+      }
+
+      if iRow >= InsideAreaNorthMinRow
+         AND iFineRow >= InsideAreaNorthMinFineRow
+         AND iRow <= InsideAreaNorthMaxRow
+         AND iFineRow <= InsideAreaNorthMaxFineRow
+         AND iCol >= InsideAreaNorthMinCol
+         AND iFineCol >= InsideAreaNorthMinFineCol
+         AND iCol <= InsideAreaNorthMaxCol
+         AND iFineCol <= InsideAreaNorthMaxFineCol
+      {
+         return AREA_INSIDE_NORTH;
+      }
+      
+      if Send(who,@GetHeightAtObject) = 286
+         OR Send(who,@GetHeightAtObject) = 290
+         OR Send(who,@GetHeightAtObject) = 924
+      {
```

Comment:
You really should have a look at the new GetLocationInfoBSP()...
There is a LOT BETTER way to determine where you are than referring it hardcoded from heights.

Just assign some "ID" to the sectors in roomeditor. Then use this in KOD:

>> iQflags = LIQ_GET_SECTORINFO;

>> GetLocationInfoBSP(prmRoom, iQflags, iMRow,iMCol,iMRowFine,iMColFine, *iRflags, *iHeightF, *iHeightFWD, *iHeightC, *iServerID)

Afterwards your local variable named "iServerID" will contain the value you see in roomeditor assigned to the sector as ID....